modules/biscuit/pileupsomatic/main.nf

```diff
@@ -0,0 +1,45 @@
+process BISCUIT_PILEUPSOMATIC {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::biscuit=1.0.2.20220113 bioconda::samtools=1.15" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-db16f1c237a26ea9245cf9924f858974ff321d6e:17fa66297f088a1bc7560b7b90dc273bf23f2d8c-0':
+        'quay.io/biocontainers/mulled-v2-db16f1c237a26ea9245cf9924f858974ff321d6e:17fa66297f088a1bc7560b7b90dc273bf23f2d8c-0' }"
+
+    input:
+    tuple val(meta), path(normal_bam), path(tumor_bam), path(normal_bai), path(tumor_bai)
```

Comment:
I see that the command is just `biscuit pileup`, can you make it more generic by just passing optional input for `tumor_ba*` etc?

so insert the `tumor_*` files only if they are _not_ an empty list `[]` via ` def` in the script block?

Reply:
I think this could be made to work but it might get pretty ugly pretty fast. If the `-S` flag is set, we need exactly one `tumor_bam` (following the `-T` flag) and exactly one `normal_bam` (following the `-I` flag). These use cases and input requirements seemed different enough to me to split into separate tools. Strelka does similarly.